sparse/numba_backend/tests/test_coo.py

```diff
@@ -1898,7 +1904,7 @@ def data_rvs(size):
     s1 = sparse.random(shape1, density=density, data_rvs=data_rvs)
     s2 = sparse.random(shape2, density=density, data_rvs=data_rvs)
 
-    axis = rng.integers(max(s1.ndim, s2.ndim))
+    axis = rng.integers(min(s1.ndim, s2.ndim))
```

Comment:
I think I would prefer to make `axis` a test parameter (`"shape1, shape2, axis"`) instead of a random integer (how do we know whether `pytest.raises(ValueError, match="Shapes must match along")` is run for every test case?).

Reply:
Done in https://github.com/pydata/sparse/pull/691/commits/a9c01db04424b137c8167316e56391d8c6dcb759. I also fixed a few bugs relating to the array API suite, but triggering them requires https://github.com/data-apis/array-api-tests/pull/272.